.circleci/setup.sh

```diff
@@ -7,8 +7,21 @@ WORKSPACE=`pwd`
 # This file can be used to set BIOCONDA_UTILS_TAG and MINICONDA_VER.
 source .circleci/common.sh
 
+# make sure the CircleCI config is up to date
+git remote add -t master upstream https://github.com/bioconda/bioconda-recipes.git
+git fetch upstream
+if ! git diff --quiet HEAD...upstream/master -- .circleci/; then
+    echo 'The CI configuration is out of date.'
+    echo 'Please merge in bioconda:master.'
+    exit 1
+fi
+
 # Set path
-echo "export PATH=$WORKSPACE/miniconda/bin:$PATH" >> $BASH_ENV
+cat >> $BASH_ENV << EOF
+if ! [[ "\$PATH" =~ (^|:)"$WORKSPACE/miniconda/bin"(|/)(:|\$) ]]; then
```

Comment:
Can you please add a comment that explains what the condition is supposed to check :-)?

Reply:
How is this different from what we had before?

Reply:
Previously we reset `PATH` unconditionally so that any `bash` script would reset it when it automatically sourced `$BASH_ENV`. E.g., with a `post-link.sh` script the activated environment would have been be kicked out of `PATH` since it would run `$BASH_ENV` which would've reset `PATH`.

Reply:
Is this a particular behavior of post-link? In general, non-interactive shells should not source .bashrc again (which is $BASH_ENV on circle ci).

Reply:
Ah, right, I think that was only a brain fart on my end. Yeah, `$BASH_ENV` shouldn't be run because we won't be using interactive shells. I'll just revert that change.